core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -817,12 +817,20 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     logger.info(s"Refreshed index searchers to make reads consistent for dataset=$ref shard=$shardNum")
   }
 
+  //scalastyle:off method.length
   private def leafFilter(column: String, filter: Filter): Query = {
     filter match {
       case EqualsRegex(value) =>
         val regex = removeRegexAnchors(value.toString)
-        if (regex.nonEmpty) new RegexpQuery(new Term(column, regex), RegExp.NONE)
-        else leafFilter(column, NotEqualsRegex(".+"))  // value="" means the label is absent or has an empty value.
+        if(regex.r.unapplySeq("").isDefined) {
```

Comment:
Nitpick: does `matches` work here?

Reply:
That's exactly what I was looking for, but its available in Scala 2.13